lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -1440,23 +1434,23 @@ llvm::Error Caffe2ModelLoader::loadWeight(const caffe2::OperatorDef &op) {
         continue;
       }
       auto dim = getShape(dict["shape"]);
-
-      T.reset(ElemKind::Int8QTy, dim, 0.0, 0);
-      auto TH = T.getHandle<int8_t>();
+      T.reset(ElemKind::UInt8QTy, dim, 0.0, 0);
+      auto TH = T.getHandle<uint8_t>();
       RETURN_ERR_IF_NOT(
           dict["values"]->strings().size() == 1,
           "Expect single string input for GivenTensorByteStringToUInt8Fill");
-      const std::string str = dict["values"]->strings().Get(0);
+      const std::string &str = dict["values"]->strings().Get(0);
 
-      // We're loading unsigned data into Int8QTy, so we use OFFSETSHIFT to
-      // convert to signed.
-      size_t i;
-      for (i = 0; i < str.size(); i++) {
-        TH.raw(i) = (uint8_t)str.c_str()[i] - OFFSETSHIFT;
+      size_t pos;
+      for (pos = 0; pos < str.size(); pos++) {
+        TH.raw(pos) = (uint8_t)str.c_str()[pos];
```

Comment:
@rdzhabarov Is it guaranteed that str.c_str() does not allocate anything? I'd suggest hoisting `std.c_str()` out of the loop, just in case.

Reply:
Why use `c_str` at all here?  It seems like `(uint8_t)str[pos]` is equivalent.

Reply:
copy pasted from above. But yeah, @bertmaher suggestion should work.